pkg/chunk/cache/instrumented.go

```diff
@@ -52,27 +53,30 @@ type instrumentedCache struct {
 	Cache
 }
 
-func (i *instrumentedCache) StoreChunk(ctx context.Context, key string, buf []byte) error {
+func (i *instrumentedCache) Store(ctx context.Context, key string, buf []byte) error {
 	return instr.TimeRequestHistogram(ctx, i.name+".store", requestDuration, func(ctx context.Context) error {
-		return i.Cache.StoreChunk(ctx, key, buf)
+		sp := ot.SpanFromContext(ctx)
+		sp.LogFields(otlog.String("key", key))
```

Comment:
This span will be a bit spammy on the traces, no?  Perhaps add tracing to the tiered cache writeback?

Reply:
Hmm, this means we cannot get the timings either. `TimeRequestHistogram` also accounts for the prometheus histogram...

Reply:
We can time separately, there is nothing stopping us.

Reply:
I'm not super happy with how I got it to work, suggestions welcome.